src/appleseed.python/TXConverter.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+class TXConverter:
+    def __init__(self, maketx_path):
+        self.converted = {}
+        self.maketx_path = maketx_path
+
+    def convert(self, path):
+        if path in self.converted:
+            return self.converted[path]
+        else:
+            path_converted = self._convert(path)
+            self.converted[path] = path_converted
+            return path_converted
+
+    def _convert(self, path):
```

Comment:
I find it slightly confusing to have two methods with the same name but with different visibility. Maybe `_do_convert()`?